cmd/tailscale/cli/debug.go

```diff
@@ -451,6 +457,64 @@ func runLocalCreds(ctx context.Context, args []string) error {
 	return nil
 }
 
+func looksLikeHTTPMethod(s string) bool {
+	if len(s) > len("OPTIONS") {
+		return false
+	}
+	for _, r := range s {
+		if r < 'A' || r > 'Z' {
+			return false
+		}
+	}
+	return true
+}
+
+func runRawLocalAPI(ctx context.Context, args []string) error {
+	if len(args) == 0 {
+		return errors.New("expected at least one argument")
+	}
+	method := "GET"
+	if looksLikeHTTPMethod(args[0]) {
+		method = args[0]
+		args = args[1:]
+		if len(args) == 0 {
+			return errors.New("expected at least one argument after method")
+		}
+	}
+	path := args[0]
+	if !strings.HasPrefix(path, "/localapi/") {
+		if !strings.Contains(path, "/") {
+			path = "/localapi/v0/" + path
+		} else {
+			path = "/localapi/" + path
+		}
+	}
+
+	var body io.Reader
+	if len(args) > 1 {
+		body = strings.NewReader(args[1])
+	}
+	req, err := http.NewRequest(method, "http://local-tailscaled.sock"+path, body)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(Stderr, "# doing request %s %s\n", method, path)
+
+	res, err := localClient.DoLocalRequest(req)
+	if err != nil {
+		return err
+	}
+	if res.StatusCode != http.StatusOK {
+		fmt.Fprintf(Stderr, "# Response status %s\n", res.Status)
+	}
+	if res.StatusCode >= 200 && res.StatusCode <= 299 {
+		io.Copy(Stdout, res.Body)
+		return nil
+	}
+	res.Write(Stdout)
```

Comment:
I dislike the change in format for 4xx errors, but I like that it's an option. `curl` and `httpie` use `-v` flags for to opt into headers. Otherwise I think `# Response status 400` to stderr then resulting body to stdout still feels right to me.

Reply:
changed, added `-v`